src/library/export/trackexportdlg.cpp

```diff
@@ -48,6 +52,36 @@ void TrackExportDlg::showEvent(QShowEvent* event) {
         qDebug() << "Programming error: did not initialize m_exporter, about to crash";
         return;
     }
+
+    //Checks if there are any files that are missing. If so, it will throw dialog informing the user
+    QList<QString> files = m_worker->getMissingTracks();
+    if (files.length() > 0) {
+        QDialog* notExportedDlg = new QDialog(nullptr);
+        notExportedDlg->setWindowTitle(tr("Export Track Files"));
+        QVBoxLayout* notExportedLayout = new QVBoxLayout;
+        QLabel* notExportedLabel = new QLabel;
+        notExportedLabel->setText(
+                tr("The following %1 files were not found at the specified "
+                   "file location and as a result were not exported")
+                        .arg(QString::number(files.length())));
+        notExportedLabel->setTextFormat(Qt::RichText);
+
+        QListWidget* notExportedList = new QListWidget;
+        notExportedList->addItems(files);
+
+        QDialogButtonBox* exportedDlgButtons = new QDialogButtonBox();
+        QPushButton* okBtn = exportedDlgButtons->addButton(
+                tr("OK"),
+                QDialogButtonBox::AcceptRole);
+        connect(okBtn, &QPushButton::clicked, notExportedDlg, &QDialog::accept);
+
+        notExportedLayout->addWidget(notExportedLabel);
+        notExportedLayout->addWidget(notExportedList);
+        notExportedLayout->addWidget(exportedDlgButtons);
+        notExportedDlg->setLayout(notExportedLayout);
+        notExportedDlg->exec();
+    }
```

Comment:
notExportedDlg is leaking here. Why do you allocate this modal dialog on the heap and not on the stack?

Reply:
I was following the code structure in the wTrackMenu.cpp file referenced above. By placing the dialog before start(), I was able to get the dialog box to work. This is my first time ever working with Qt and C++ so I'm not sure what my options are in terms of placing this code block, or what you mean by allocating in heap vs. stack (I know what those two data structures are, but don't know how they apply in this context)

Reply:
If you program in C/C++ you first need to get familiar with how memory is managed. Operator `new` allocates memory dynamically on the heap that needs to be freed eventually.